src/addons/blog/services/blog.ts

```diff
@@ -183,6 +275,102 @@ export class AddonBlogProvider {
         return site.write('core_blog_view_entries', data);
     }
 
+    /**
+     * Format local stored entries to required data structure.
+     *
+     * @param offlineEntry Offline entry data.
+     * @param entry Entry.
+     * @returns Formatted entry.
+     */
+    async formatOfflineEntry(
+        offlineEntry: AddonBlogOfflineEntry,
+        entry?: AddonBlogPostFormatted,
+    ): Promise<AddonBlogOfflinePostFormatted> {
+        const options: AddonBlogAddEntryOption[] = JSON.parse(offlineEntry.options);
+        const moduleid = options?.find(option => option.name === 'modassoc')?.value as number | undefined;
+        const courseid = options?.find(option => option.name === 'courseassoc')?.value as number | undefined;
+        const tags = options?.find(option => option.name === 'tags')?.value as string | undefined;
+        const publishstate = options?.find(option => option.name === 'publishstate')?.value as AddonBlogPublishState
+            ?? AddonBlogPublishState.draft;
+        const user = await CoreUser.getProfile(offlineEntry.userid);
+        const offlineFiles = await AddonBlogOffline.getOfflineFiles(offlineEntry.id ?? offlineEntry.created);
+        const optionsFiles = this.getAttachmentFilesFromOptions(options);
+        const attachmentfiles = [...optionsFiles.online, ...offlineFiles];
```

Comment:
```suggestion
        const moduleId = options?.find(option => option.name === 'modassoc')?.value as number | undefined;
        const courseId = options?.find(option => option.name === 'courseassoc')?.value as number | undefined;
        const tags = options?.find(option => option.name === 'tags')?.value as string | undefined;
        const publishState = options?.find(option => option.name === 'publishstate')?.value as AddonBlogPublishState
            ?? AddonBlogPublishState.draft;
        const user = await CoreUser.getProfile(offlineEntry.userid);
        const offlineFiles = await AddonBlogOffline.getOfflineFiles(offlineEntry.id ?? offlineEntry.created);
        const optionsFiles = this.getAttachmentFilesFromOptions(options);
        const attachmentFiles = [...optionsFiles.online, ...offlineFiles];
```
Our variables should use camelCase, unless when we're using destructuring and the name comes from a WebService response.